File: ruby/worker.rb
require 'beaneater'
require 'json'

@beanstalk = Beaneater::Pool.new(['localhost:11300'])
# Enqueue jobs to tube
@tube = @beanstalk.tubes["my-tube"]

while @tube.peek(:ready)
  job = @tube.reserve
  puts "job value is #{JSON.parse(job.body)["key"]}!"
  job.delete
end
# Disconnect the pool
@beanstalk.close

File: c++/worker.cpp
#include "beanstalk.hpp"
#include <iostream>
#include <assert.h>

using namespace std;
using namespace Beanstalk;

int main() {
    Client client("127.0.0.1", 11300);
    assert(client.use("test"));
    assert(client.watch("test"));

    Job job;

    for (;;) {

      assert(client.reserve(job) && job);

      cout << "reserved job id: "
           << job.id()
           << " with body {" << job.body() << "}"
           << endl;

      assert(client.del(job.id()));
      cout << "deleted job id: " << job.id() << endl;
    }
}
